### PR TITLE
Show Page Links

### LINKED ISSUE #5: User Story 5, All user names are links to a show page
```
As a visitor
With the exception of a user's show page,
Anywhere I see a user's name on the site for a book review,
I can click on the name to go to that user's show page.
```

DO THESE STORIES FIRST:
- User Story 16, User Show Page

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [ ] edge case ("sad path") tests written
- [ ] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #31

child of #33

depends on #11

### PR DESCRIPTION
**Pull request by alexander-mathieu**
This PR turns all author, coauthor and user names into links throughout the app.  To accomplish this:

- Book model coauthor method was refactored to return an array of objects rather than an array of strings
- Book model author_names method was removed, along with all dependent tests
- Author/coauthor appearances have been converted to unordered list items rather than comma separated values

Resolves #5 , #6

### CODE REVIEW

File: app/views/authors/show.html.erb [hunk 1]
```diff
@@ -7,18 +7,22 @@
     <p>Pages: <%= book.pages %></p>
     <p>Published: <%= book.year %></p>
     <% if book.coauthors(@author).any? %>
-      <p>Co-authors: <%= book.coauthors(@author).join(', ') %></p>
+      <p>Co-authors:</p>
+        <% book.coauthors(@author).each do |coauthor| %>
+        <ul>
+          <li><%= link_to coauthor.name, author_path(coauthor) %></li>
+        </ul>
+        <% end %>
```
Comment (BrennanAyers) [inline app/views/authors/show.html.erb:15]: 👍

File: app/models/book.rb [hunk 2]
```diff
@@ -8,10 +8,6 @@ class Book < ApplicationRecord
   validates :title, :pages, :year, presence: true
   validates :title, uniqueness: true
 
-  def author_names
-    authors.pluck(:name)
-  end
-
```
Comment (BrennanAyers) [inline app/models/book.rb:14]: 😄
